src/tsickle.ts

```diff
@@ -2050,18 +2050,23 @@ function addClutzAliases(
       // https://github.com/angular/clutz/tree/master/src/test/java/com/google/javascript/clutz
       const clutzSymbolName = 'module$contents$' + clutzModuleName + '_' + symbol.name;
 
+      const genericParams = getGenericTypeParameters(symbol);
+
       if (symbol.flags & ts.SymbolFlags.Class) {
         // classes need special care to match clutz, which seperates class types into a
         // type for the static properties and a type for the instance properties
-        reexports += `\t\ttype ${clutzSymbolName} = ${symbol.name};\n`;
+        reexports += `\t\ttype ${clutzSymbolName}${genericParams.constrainedParams} = ${
```

Comment:
This is getting gnarly, consider helper methods like emit, indent,unindent. `https://github.com/angular/clutz/blob/master/src/main/java/com/google/javascript/clutz/DeclarationGenerator.java#L1129`.

Could be done in the next PR.

Reply:
Made a partial step in that direction.